docs/advanced-customization/custom-templates.md

```diff
@@ -748,6 +749,73 @@ The following props are passed to each `UnsupportedFieldTemplate`:
 - `reason`: The reason why the schema field has an unsupported type.
 - `registry`: The `registry` object.
 
+## WrapIfAdditionalTemplate
+
+The `WrapIfAdditionalTemplate` is used by the FieldTemplate to conditionally render additional controls if `additionalProperties` is present in the schema. You may customize `WrapIfAdditionalTemplate` if you wish to change the layout or behavior of user-controlled `additionalProperties`.
+
+```tsx
+import { WrapIfAdditionalTemplateProps } from "@rjsf/utils";
+
+function WrapIfAdditionalTemplate<T = any, F = any>(
+  props: WrapIfAdditionalTemplateProps<T, F>
+) {
+  const {
+    id,
+    label,
+    onKeyChange,
+    onDropPropertyClick,
+    schema,
+    children,
+    uiSchema,
+    registry,
+  } = props;
+  const uiOptions = getUiOptions(uiSchema);
+  const { RemoveButton } = getTemplate("ButtonTemplates", registry, uiOptions);
```

Comment:
Button templates are not overridable in uiOptions so you can simplify it to this
```suggestion
  const { RemoveButton } = registry.templates.ButtonTemplates;
```